app/services/notifications/consolidation_planner.rb

```diff
@@ -98,6 +99,35 @@ def liked_by_two_users
         )
     end
 
+    def linked
+      ConsolidateNotifications
+        .new(
+          from: Notification.types[:linked],
+          to: Notification.types[:linked_consolidated],
+          threshold: -> { SiteSetting.notification_consolidation_threshold },
+          consolidation_window: SiteSetting.linked_notification_consolidation_window_mins.minutes,
+          unconsolidated_query_blk:
+            Proc.new do |notifications, data|
+              key = "display_username"
+              value = data[key.to_sym]
+              filtered = notifications.where("data::json ->> 'username2' IS NULL")
+
+              filtered = filtered.where("data::json ->> '#{key}' = ?", value) if value
+
+              filtered
+            end,
+          consolidated_query_blk: filtered_by_data_attribute("display_username"),
+        )
+        .set_mutations(
+          set_data_blk:
+            Proc.new do |notification|
+              data = notification.data_hash
+              data.merge(username: data[:display_username])
+            end,
+        )
+        .set_precondition(precondition_blk: Proc.new { |data| data[:username2].blank? })
```

Comment:
Does `username2` here have to do with two different people linking to the same topic/post?

Reply:
I'll look at this again, we likely don't need this for links since they aren't as common as likes. I got this block of code from: https://github.com/discourse/discourse/blob/6cf7447cb90b0ca632747d56ca2f5f0eed06d15c/app/services/notifications/consolidation_planner.rb#L27-L54

And I think username2 is coming from: https://github.com/discourse/discourse/blob/6cf7447cb90b0ca632747d56ca2f5f0eed06d15c/app/assets/javascripts/discourse/app/lib/notification-types/liked.js#L34